lib/logsource.c

```diff
@@ -53,17 +53,23 @@ log_source_window_empty(LogSource *self)
 }
 
 static inline void
-_flow_control_window_size_adjust(LogSource *self, guint32 window_size_increment)
+_flow_control_window_size_adjust(LogSource *self, guint32 window_size_increment, gboolean last_ack_type_is_suspended)
 {
-  guint32 old_window_size;
-
-  window_size_increment += g_atomic_counter_get(&self->suspended_window_size);
-  old_window_size = g_atomic_counter_exchange_and_add(&self->window_size, window_size_increment);
-  g_atomic_counter_set(&self->suspended_window_size, 0);
-
-  if (old_window_size == 0)
+  gboolean suspended;
+  gsize old_window_size = window_size_counter_add(&self->window_size, window_size_increment, &suspended);
+
+  msg_debug("window_size_adjust", evt_tag_int("window_size",
+                                              old_window_size+window_size_increment),
+            evt_tag_str("suspended:", suspended ? "TRUE" : "FALSE"),
+            evt_tag_str("AT_SUSPENDED:", last_ack_type_is_suspended ? "TRUE" : "FALSE"));
```

Comment:
Would you like to leave this debug message in the final patch?

Reply:
Well.. I found it very useful during development... but it is possible to print out a better message, I'll try to do that. But I'd keep a debug message here.